server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -742,6 +743,9 @@ public BlobStore blobStore() {
                     }
                     try {
                         store = createBlobStore();
+                        if (Boolean.TRUE.equals(metadata.encrypted())) {
+                            store = new EncryptedBlobStore(store, metadata.cryptoMetadata());
+                        }
```

Comment:
As mentioned in the other comment, lets avoid two properties, metadata's presence should be enough to consider repo being encrypted

Reply:
Pushed this change.